dms/models/directory.py

```diff
@@ -265,11 +265,15 @@ def _get_own_root_directories(self, user_id):
     def check_access(self, operation, raise_exception=False):
         res = super(DmsDirectory, self).check_access(operation, raise_exception)
         if self.env.user.has_group("base.group_portal"):
-            if self.id in self._get_ids_without_access_groups(operation):
+            if self.ids in self._get_ids_without_access_groups(operation):
```

Comment:
This is probably a bug.

Both `self.ids` and `self._get_ids_without_access_groups(operation)` return `list` objects, and the code seems to try to act as if they were `set`s.

See:

```python
>>> a=[1,2]
>>> b=[2,3,1]
>>> a in b
False
>>> c={1,2}
>>> d={2,3,1}
>>> d in c
False
>>> d & c
{1, 2}
>>> bool(d & c)
True
```

Fix: 

```suggestion
            if set(self.ids) & set(self._get_ids_without_access_groups(operation)):
```

Reply:
Interesting explanation about it, i have changed some things similar.